src/platform/utilities/ui/focus.js

```diff
@@ -39,6 +47,14 @@ export function focusElement(selectorOrElement, options, root) {
     }
     el.focus(options);
   }
+
+  if (el) {
+    if (isWebComponent(el) && !isWebComponentReady(el)) {
+      awaitShadowRoot(el, focus);
+    } else {
```

Comment:
Since I've been using the focus element function like this:

```js
focusElement('button', {}, document.querySelector('va-button').shadowRoot);
```
We need to rethink how to use this function, because using `isWebComponent(el) && !isWebComponentReady(el)` is trying to target the button inside the web component.

Reply:
right, in that case, it's actually just concerned with the button, so it'll just do a regular focus.
But yeah, I think awaiting the shadowRoot of the host is a good idea in all scenarios where we depend on it, especially as a prerequisite to querying dom inside.

I think no action is needed for this comment right?  Just something for us to consider (there may be some bugs related to assuming shadowRoot exists), and we can address in the future?

Reply:
perhaps we need another utility for that kind of dom targeting inside of a shadowRoot.

Reply:
After discussing some with Robin, going to make another change so that we can also use a focus with a root and child selector, which seems to be one of the intents of this function.